### PR TITLE
Fp 141 - filename variable optional

### PR DESCRIPTION
**Pull request by edolsen-rv**
Moved the filename to optional. Deleted the stub .zip files from the examples. All examples successfully execute a `terraform plan`

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -93,6 +90,11 @@ variable "vpc_tag_key_override" {
 # These parameters have reasonable defaults.
 # ---------------------------------------------------------------------------------------------------------------------
 
+variable "filename" {
+  description = "The path to the function's deployment package within the local filesystem."
+  default     = ""
```
Comment (lbrimeyer) [inline variables.tf]: possible to leave this null and check that it's passed at all instead of misusing a zero val?

Comment (edolsen-rv) [reply variables.tf]: will this look like `filename = var.filename != null ? var.filename : ""` ?
I'm getting an error when trying to run the examples:
```
 Error: Unsupported argument
│ 
│   on ../../main.tf line 4, in module "lambda":
│    4:   filename = var.filename != null ? var.filename : ""
│ 
│ An argument named "filename" is not expected here.
```

Comment (lbrimeyer) [reply variables.tf]: Ah so the underlying module shtuff can't handle nulls, gotcha

**Review by lbrimeyer — DISMISSED** [review]

File: variables.tf (hunk 1 above)
Comment (lbrimeyer) [reply variables.tf]: just make sure the zero value is documented that an empty string is basically psuedo-null to fulfill the interface

**Comment by edolsen-rv** [comment]
@lbrimeyer Can I get this stamped again?

**Review by lbrimeyer — APPROVED** [review]

**Pull request merged**